plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```
@@ -766,26 +766,53 @@ public KVMStoragePool createStoragePool(String name, String host, int port, Stri
         }
     }
 
+    private boolean destroyStoragePool(Connect conn, String uuid) throws LibvirtException
+    {
+        StoragePool sp;
+        try {
+            sp = conn.storagePoolLookupByUUIDString(uuid);
+        } catch (LibvirtException exc) {
+            s_logger.warn("Storage pool " + uuid + " doesn't exist in libvirt. Assuming it is already removed");
```

This is masking the exception, we should at least log it.